BadgeFramework/badge_gui_bleak.py

```diff
@@ -0,0 +1,363 @@
+import tkinter as tk
+from tkinter import ttk
+import asyncio
+import pandas as pd
+from badge import OpenBadge
+from datetime import datetime
+# import numpy as np
+import sys
+import time
+import ntplib
+# from functools import partial
+
+
+SENSOR_ALL = 0
+SENSOR_MICROPHONE = 1
+SENSOR_IMU = 2
+SENSOR_SCAN = 3
+SENSOR_CHECK_STATUS = 100
+SENSOR_TIMEOUT = 15
+
+SENSOR_START = True
+SENSOR_STOP = False
+CHECK_SYNC = True
+CHECK_NO_SYNC = False
+
+BADGES_ALL = -1000
+USE_ALL = False  # True to use all badges in .csv file and False to use only marked badges
+
+sensors = ['All', 'Mic', 'IMU', 'Scan']
+indicators_long = ['clock', 'microphone', 'imu', 'scan']
+indicators_short = ['Clock', 'Mic', 'IMU', 'Scan']
+sensor_num = len(indicators_short)
+
+
+class RedirectText:
+    """Class to redirect stdout to a tkinter Text widget."""
+
+    def __init__(self, text_widget):
+        self.text_widget = text_widget
+        self.text_widget.config(state=tk.NORMAL)
+
+    def write(self, string):
+        """Redirects the text to the Text widget."""
+        self.text_widget.insert(tk.END, string)
+        self.text_widget.see(tk.END)  # Auto-scroll to the bottom
+        self.text_widget.update_idletasks()
+
+    def flush(self):
+        """Handle flush."""
+        pass  # Needed for compatibility with the print function
+
+
+class BadgeMonitorApp(tk.Tk):
+    def __init__(self, ):
+        super().__init__()
+
+        self.title("Badge Status Monitor")
+        self.badges = pd.read_csv('mappings2.csv')
+        # self.badges = []
+
+        # Create a frame to contain the badge area and the terminal area
+        self.main_frame = ttk.Frame(self)
+        self.main_frame.grid(row=0, column=0, sticky="nsew")  # Use grid for main_frame
+
+        # Make the window resizable in both directions
+        self.grid_rowconfigure(0, weight=1)
+        self.grid_columnconfigure(0, weight=1)
+
+        # Create a canvas to allow scrolling for the badge section
+        self.canvas = tk.Canvas(self.main_frame, borderwidth=0)
+        self.frame = ttk.Frame(self.canvas)
+        self.scrollbar = ttk.Scrollbar(self.main_frame, orient="vertical", command=self.canvas.yview)
+        self.canvas.configure(yscrollcommand=self.scrollbar.set)
+
+        # Place the scrollbar and canvas in the window
+        self.scrollbar.grid(row=0, column=0, sticky="ns")
+        self.canvas.grid(row=0, column=1, sticky="nsew")
+
+        # Bind mouse scroll to canvas
```

Comment:
Please remove all comments that do not add anything to the code, as this one, which says the say as the function name